reversers/database-synchroniser/plugins/org.obeonetwork.dsl.database.liquibasegen/src/org/obeonetwork/dsl/database/liquibasegen/service/DefaultTypeMatcher.java

```diff
@@ -0,0 +1,150 @@
+ * generation
+ *
+ */
+public class DefaultTypeMatcher {
+
+	private static final String BINARY_FORMAT = "Binary";
+	private static final String BOOLEAN_FORMAT = "Bool�en";
+	private static final String DATE_FORMAT = "Date";
+	private static final String DECIMAL_FORAMT = "D�cimal";
+	private static final String INT_FORMAT = "Entier";
+	private static final String LONG_FORMAT = "Entier long";
+	private static final String FLOAT_FORMAT = "R�el";
+	private static final String STRING_FORMAT = "Texte";
+	private static final String TIME_FORMAT = "Time";
+	private static final String TIMESTAMPFORMAT = "Timestamp";
+
+	public enum LiquibaseDefaultType {
+
+		/**
+		 * Returned if the given type has an unknown data type
+		 */
+		UNKWOWN,
+		/**
+		 * Return if the model is not complete enough to guess the matching types
+		 */
+		INVALID,
+		/**
+		 * Returned for data type that can set default value in liquibase
+		 */
+		UNHANDLED, BOOLEAN, STRING, DATE, NUM
+
+	}
+
+	/**
+	 * Gets the matching {@link LiquibaseDefaultType} for the given
+	 * {@link NativeType}
+	 * 
+	 * @param nativeType a {@link NativeType}
+	 * @return a {@link LiquibaseDefaultType}
+	 */
+	public static LiquibaseDefaultType getLiquibaseDefaultType(NativeType nativeType) {
+		if (nativeType != null) {
+			NativeType mdlType = nativeType.getMapsTo();
+			if (mdlType != null) {
+				switch (mdlType.getName()) {
+				case BOOLEAN_FORMAT:
+					return LiquibaseDefaultType.BOOLEAN;
+				case DECIMAL_FORAMT:
+				case INT_FORMAT:
+				case FLOAT_FORMAT:
+				case LONG_FORMAT:
+					return LiquibaseDefaultType.NUM;
+				case TIME_FORMAT:
+				case DATE_FORMAT:
+					return LiquibaseDefaultType.DATE;
+				case STRING_FORMAT:
+					return LiquibaseDefaultType.STRING;
+				case BINARY_FORMAT:
+				case TIMESTAMPFORMAT:
+					return LiquibaseDefaultType.UNHANDLED;
+				default:
+					return LiquibaseDefaultType.UNKWOWN;
+				}
+			}
+		}
+		return LiquibaseDefaultType.INVALID;
+	}
+
+	/**
+	 * Validate the given value for the matching data type
+	 * 
+	 * @param type  a {@link LiquibaseDefaultType}
+	 * @param value the default value
+	 * @return an {@link Status#OK_STATUS} if ok, a status with a message otherwise
+	 */
+	public static IStatus validateValue(LiquibaseDefaultType type, String value) {
+		if (type == null) {
+			return Status.OK_STATUS;
+		}
+		switch (type) {
+		case DATE:
+			return validateDateDefaultValue(value);
+		default:
+			return Status.OK_STATUS;
+		}
+	}
+
+	/**
+	 * Pre process the default value to it can be used by the Liquibase API
+	 * 
+	 * @param type  the Liquibase type
+	 * @param value the input value
+	 * @return a transformed (or not) value
+	 */
+	public static String preProcessDefaultValue(LiquibaseDefaultType type, String value) {
+		if (type == null) {
+			return value;
+		}
+		switch (type) {
+		case STRING:
+			return preProcessString(value);
+		default:
+			return value;
+		}
+	}
+
+	private static String preProcessString(String value) {
```

Comment:
Don't we need to trim value before starting analyzing it ?

Reply:
Done